crates/biome_js_type_info/src/type_info.rs

```diff
@@ -0,0 +1,375 @@
+//! Data structures for tracking type information across JS/TS files and scopes.
+//!
+//! Note that while our type inference is based on TypeScript and aims to be
+//! compatible with TypeScript's types, we apply the same inference to
+//! JavaScript files as well.
+//!
+//! This can be used by type-informed lint rules such as `noFloatingPromises`.
+//!
+//! The type information is instantiated and updated inside the Workspace
+//! Server.
+
+use std::str::FromStr;
+
+use biome_rowan::Text;
+use static_assertions::assert_eq_size;
+
+/// Represents an inferred TypeScript type.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub enum Type {
```

Comment:
How are TS `interface` represented?

Reply:
They aren’t yet. There are some TODOs about it, but I think they’re in the other file where I’m matching all the variants to “infer” the `Type` from the syntax definition.